modin/pandas/utils.py

```diff
@@ -82,3 +87,101 @@ def decorator(cls):
         return cls
 
     return decorator
+
+
+# Register a fix import function to run on all_workers including the driver.
+# This is a hack solution to fix #647, #746
+def _move_stdlib_ahead_of_site_packages(*args):
+    site_packages_path = None
+    site_packages_path_index = -1
+    for i, path in enumerate(sys.path):
+        if sys.exec_prefix in path and path.endswith("site-packages"):
+            site_packages_path = path
+            site_packages_path_index = i
+            # break on first found
+            break
+
+    if site_packages_path is not None:
+        # stdlib packages layout as follows:
+        # - python3.x
+        #   - typing.py
+        #   - site-packages/
+        #     - pandas
+        # So extracting the dirname of the site_packages can point us
+        # to the directory containing standard libraries.
+        sys.path.insert(site_packages_path_index, os.path.dirname(site_packages_path))
+
+
+# Register a fix to import pandas on all workers before running tasks.
+# This prevents a race condition between two threads deserializing functions
+# and trying to import pandas at the same time.
+def _import_pandas(*args):
+    import pandas  # noqa F401
+
+
+def initialize_ray(cluster=None, redis_address=None, redis_password=None):
+    import ray
+
+    """Initializes ray based on environment variables and internal defaults."""
+    if threading.current_thread().name == "MainThread":
+        import secrets
+
+        plasma_directory = None
+        num_cpus = os.environ.get("MODIN_CPUS", None) or multiprocessing.cpu_count()
+        cluster = os.environ.get("MODIN_RAY_CLUSTER", None) or cluster
+        redis_address = os.environ.get("MODIN_REDIS_ADDRESS", None) or redis_address
```

Comment:
@devin-petersohn maybe we should change this to
```suggestion
        cluster = cluster or os.environ.get("MODIN_RAY_CLUSTER", None)
        redis_address = redis_address or os.environ.get("MODIN_REDIS_ADDRESS", None)
```
this would allow retaining both calling no-parameters `initialize_ray()` that would read environment and to use parameters to override the environment